ui/ui.go

```diff
@@ -52,6 +52,19 @@ type UI struct {
 	state              uint64
 	minlevel           Level
 	progressBarEnabled bool
+	exit               func(int)
+}
+
+func (w *UI) SetExit(f func(int)) {
+	w.exit = f
+}
+
+func (w *UI) Exit(c int) {
+	if w.exit == nil {
+		os.Exit(c)
+	} else {
+		w.exit(c)
+	}
 }
```

Comment:
This is fine, but seems unnecessary: you could just set `UI.exit = os.Exit` in `New()`, and remove this function altogether.